test/e2e/tests/test_notebook_instance_lifecycle.py

```diff
@@ -0,0 +1,146 @@
+import random
+
+from e2e.replacement_values import REPLACEMENT_VALUES
+from time import sleep
+
+DELETE_WAIT_PERIOD = 16
+DELETE_WAIT_LENGTH = 30
+
+
+@pytest.fixture(scope="function")
+def notebook_instance_lifecycleConfig():
+    notebook_instance_lfc_name = random_suffix_name("notebookinstancelfc", 32)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["NOTEBOOK_INSTANCE_LFC_NAME"] = notebook_instance_lfc_name
+    reference, spec, resource = create_sagemaker_resource(
+        resource_plural="notebookinstancelifecycleconfigs",
+        resource_name=notebook_instance_lfc_name,
+        spec_file="notebook_instance_lifecycle_config",
+        replacements=replacements,
+    )
+    assert resource is not None
+    yield (reference, resource, spec)
+
+    if k8s.get_resource_exists(reference):
+        _, deleted = k8s.delete_custom_resource(
+            reference, DELETE_WAIT_PERIOD, DELETE_WAIT_LENGTH
+        )
+        assert deleted
+
+
+def get_notebook_instance_lifecycle_config(notebook_instance_lfc_name: str):
+    try:
+        resp = sagemaker_client().describe_notebook_instance_lifecycle_config(
+            NotebookInstanceLifecycleConfigName=notebook_instance_lfc_name
+        )
+        return resp
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a Notebook Instance Lifecycle Configuration with the name {notebook_instance_lfc_name}. Error {error}"
+        )
+        return None
+
+
+@service_marker
+@pytest.mark.canary
+class TestNotebookInstanceLifecycleConfig:
+    def wait_until_update(
+        self, reference, previous_modified_time, wait_period=10, wait_time=5
+    ):
+        for i in range(wait_period):
+            resource = k8s.get_resource(reference)
+            assert resource is not None
+            assert "lastModifiedTime" in resource["status"]
+            lastModifiedTime = resource["status"]["lastModifiedTime"]
+            d1 = datetime.datetime.strptime(lastModifiedTime, "%Y-%m-%dT%H:%M:%SZ")
+            d2 = datetime.datetime.strptime(
+                previous_modified_time, "%Y-%m-%dT%H:%M:%SZ"
+            )
+            if d1 > d2:
+                return True
+            sleep(wait_time)
+        return False
+
+    def test_CreateUpdateDeleteNotebookLifecycleConfig(
+        self, notebook_instance_lifecycleConfig
+    ):
+        (reference, resource, spec) = notebook_instance_lifecycleConfig
+        assert k8s.get_resource_exists(reference)
+
+        # Getting the resource name
+        notebook_instance_lfc_name = resource["spec"].get(
+            "notebookInstanceLifecycleConfigName", None
+        )
+        assert notebook_instance_lfc_name is not None
+        current_time = datetime.datetime.today()
+        # Verifying that its set correctly
+        notebook_instance_lfc_desc = get_notebook_instance_lifecycle_config(
+            notebook_instance_lfc_name
+        )
+        assert (
```

Comment:
can remove this assert, not really useful to assert one field in spec

we need a larger change to validate the whole spec

Reply:
removed it